backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/shipment/M_InOut_StepDef.java

```diff
@@ -233,36 +233,36 @@ public void invokeGenerateShipmentsProcess(
 			final boolean isShipToday,
 			@NonNull final DataTable table)
 	{
-		final List<Map<String, String>> dataTable = table.asMaps();
-		invokeGenerateShipmentsProcess0(quantityType, isCompleteShipments, isShipToday, dataTable);
+		invokeGenerateShipmentsProcess0(quantityType, isCompleteShipments, isShipToday, DataTableRows.of(table));
 	}
 
 	public void invokeGenerateShipmentsProcess0(
 			@NonNull final String quantityType,
 			final boolean isCompleteShipments,
 			final boolean isShipToday,
-			@NonNull final List<Map<String, String>> dataTable)
+			@NonNull final DataTableRows dataTable)
 	{
 		final ImmutableMap.Builder<ShipmentScheduleId, StockQtyAndUOMQty> qtysToDeliverOverride = ImmutableMap.builder();
 		final ImmutableList.Builder<ShipmentScheduleId> schedIdsToEnqueue = ImmutableList.builder();
-		for (final Map<String, String> tableRow : dataTable)
-		{
-			final String shipmentScheduleIdentifier = DataTableUtil.extractStringForColumnName(tableRow, I_M_ShipmentSchedule.COLUMNNAME_M_ShipmentSchedule_ID + "." + TABLECOLUMN_IDENTIFIER);
-			final I_M_ShipmentSchedule shipmentSchedule = InterfaceWrapperHelper.create(shipmentScheduleTable.get(shipmentScheduleIdentifier), I_M_ShipmentSchedule.class);
-			final ShipmentScheduleId shipmentScheduleId = ShipmentScheduleId.ofRepoId(shipmentSchedule.getM_ShipmentSchedule_ID());
-			schedIdsToEnqueue.add(shipmentScheduleId);
-
-			final BigDecimal qtyToDeliverOverride = DataTableUtil.extractBigDecimalOrNullForColumnName(tableRow, ShipmentScheduleEnqueuer.ShipmentScheduleWorkPackageParameters.PARAM_QtyToDeliver_Override);
-			if (qtyToDeliverOverride != null)
-			{
-				final ProductId productId = ProductId.ofRepoId(shipmentSchedule.getM_Product_ID());
-				final I_M_Product productRecord = productDAO.getById(productId);
-				qtysToDeliverOverride.put(shipmentScheduleId, StockQtyAndUOMQty.builder()
-						.productId(productId)
-						.stockQty(Quantitys.of(qtyToDeliverOverride, UomId.ofRepoId(productRecord.getC_UOM_ID())))
-						.build());
-			}
-		}
+		dataTable.forEach(tableRow ->
+				{
+					final StepDefDataIdentifier shipmentScheduleIdentifier = tableRow.getAsIdentifier(I_M_ShipmentSchedule.COLUMNNAME_M_ShipmentSchedule_ID);
+					final I_M_ShipmentSchedule shipmentSchedule = InterfaceWrapperHelper.create(shipmentScheduleTable.get(shipmentScheduleIdentifier), I_M_ShipmentSchedule.class);
+					final ShipmentScheduleId shipmentScheduleId = ShipmentScheduleId.ofRepoId(shipmentSchedule.getM_ShipmentSchedule_ID());
+					schedIdsToEnqueue.add(shipmentScheduleId);
+
+					tableRow.getAsOptionalBigDecimal(ShipmentScheduleEnqueuer.ShipmentScheduleWorkPackageParameters.PARAM_QtyToDeliver_Override)
+							.ifPresent(qtyToDeliverOverride ->
+							{
+								final ProductId productId = ProductId.ofRepoId(shipmentSchedule.getM_Product_ID());
+								final I_M_Product productRecord = productDAO.getById(productId);
+								qtysToDeliverOverride.put(shipmentScheduleId, StockQtyAndUOMQty.builder()
```

Comment:
why not using de.metas.quantity.StockQtyAndUOMQtys#ofQtyInStockUOM(java.math.BigDecimal, de.metas.product.ProductId) ?

Reply:
Didn't actively read this part, only switched from map to datatablerows. Changing now